src/sections/trade/sections/otc/orders/OtcOrders.tsx

```diff
@@ -54,8 +60,20 @@ export const OtcOrderTable = ({ data, showMyOrders, showPartial }: Props) => {
     if (showMyOrders) {
       res = res.filter((o) => o.owner === userAddress)
     }
+
+    if (searchVal) {
```

Comment:
could you use arraySearch fn?

Reply:
I don't think arraySearch in helpers.tsx as it does not handle nested objects search. The data structure in OtcOrders.tsx was built in a way that the underlying assets are nested under offer and accepting and this makes it incompatible with arraySearch. I could rewrite the code to make it compatible but this felt like too much overreach into a codebase that I am still learning and I wanted to isolate these UI changes to have minimal impact on the rest of the base. 

If this approach does not work I can dive deeper into the helper function to modify it. Please let me know which approach you'd like me to use.

Reply:
okay then. I think it's unnecessary to have a helper fn for nested objects since we don't use them